### DISCUSSION TITLE
Persistent, dynamic buttons

### DISCUSSION THREAD

**Discussion by sendize** [Rapptz/discord.py#9851] [Q&A] [answered]
Hello!

I need a custom_id per button per message. I have to parse the custom_id so the response to each message is different. Naturally, I'd want this button to be persistent. There is an example on the repo for a [DynamicButton/DynamicItem](https://github.com/Rapptz/discord.py/blob/master/examples/views/persistent.py#L2) that fits my use-case but it doesn't seem to be included in the current release (#9700).

Is there an equivalent for this in the current release? Or perhaps other ways to achieve the same thing? I tried [`add_item` per custom_id](https://stackoverflow.com/questions/76350295/how-do-i-make-dynamically-persistant-buttons) (like below) in the `setup_hook` but it seems to be limited to 102 items per view (seems to work though if less than 102). I'm thinking of just adding more views (so a view per 100 items) but I don't think that's the proper approach.

```python
class MyBot(commands.Bot):
    ...

    async def setup_hook(self):
        view = MyView()

        for custom_id in custom_ids:
            view.add_item(MyButton(custom_id))

        self.add_view(view)

    ...
```

Any help or insight is appreciated!

**Comment by Rapptz** [comment]
I would suggest just waiting until dynamic items get released.

**Reply by sendize** [reply]
Appreciate the response! Any timeline or expected dates you could share for when that gets released?

**Reply by Rapptz** [reply]
There are a few PRs blocking release but it should be somewhat soon. I don't like making promises or ETAs though.

**Reply by sendize** [reply]
Understandable, thanks! Will wait for the new release.

**Comment by sendize** [comment, marked as answer]
`DynamicItem`s have been released on 2.4.0. Haven't personally tried it since we had to change library due to time constraints, but this would've solved our issue.
